### ISSUE TITLE
RuntimeError: location for one or more nodes not found in node location index 

### ISSUE THREAD

**Issue by NaniBird** [osmcode/libosmium#251]
Hi,

I ran the `osmium_road_length` example on 4 different osm files, and I always get the same following error:

```
terminate called after throwing an instance of 'osmium::not_found'
  what():  location for one or more nodes not found in node location index
Aborted (core dumped)
```

Thank you.

**Comment by joto** [comment]
Then your OSM files do not contain all the nodes they need for the ways.

**Comment by joto** [comment]
No, there is no way to do this. In all cases I have seen the nodes are missing because somebody created a data extract in the wrong way. Deleting the ways would delete data that is actually there. The fix for this is to create the extract in the right way. (See the `osmium extract` manpage for the different ways of creating extracts, other software uses similar techniques.)

Can you explain why, in your use case, deleting this data is okay?
